### ISSUE TITLE
Can't install 2.160 due to conflicts

### ISSUE THREAD

**Issue by danielgindi** [fastlane/fastlane#17416]
### New Issue Checklist

- [~] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

For quite some time I tried to update *fastlane*, but got completely ignored after long minutes of waiting for `sudo gem install fastlane`.

So I got sick of those new version notifications from *fastlane*, and tried to uninstall it completely, clear caches, and install.  
Got the same result.

So I tried `sudo gem install fastlane -v 2.160.0`.

Now I got this interesting thing. `conflicting dependencies faraday (~> 0.12) and faraday (~> 1.0)`.
Basically a *fastlane* dependency has a dependency which conflicts with *fastlane* 2.160.0. And this is not npm, so it can't handle it...

##### Command executed
`sudo gem install fastlane -v 2.160.0`

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>
    conflicting dependencies faraday (~> 0.12) and faraday (~> 1.0)
  Activated faraday-1.0.0
  which does not match conflicting dependency (~> 0.12)

  Conflicting dependency chains:
    fastlane (= 2.160.0), 2.160.0 activated, depends on
    faraday (~> 1.0), 1.0.0 activated

  versus:
    fastlane (= 2.160.0), 2.160.0 activated, depends on
    google-api-client (>= 0.37.0, < 0.39.0), 0.37.0 activated, depends on
    googleauth (~> 0.9), 0.9.0 activated, depends on
    faraday (~> 0.12)

  Gems matching faraday (~> 0.12):
    faraday-0.17.0, faraday-0.15.4
</pre>
</details>

### Environment

<details>
  <pre><details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.7                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.24.3 (Apple Git-128)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.15.7 (19H2)                                        |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 12.0.1                                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.152.0 | 🚫 Update available |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.1        |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.3        |
| claide                    | 1.0.3        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.14.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.6.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.7.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.7.0        |
| tty-cursor                | 0.7.0        |
| tty-spinner               | 0.9.1        |
| babosa                    | 1.0.3        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.71.1       |
| unf_ext                   | 0.0.7.6      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday                   | 0.17.0       |
| faraday_middleware        | 0.13.1       |
| fastimage                 | 2.1.7        |
| gh_inspector              | 1.1.3        |
| json                      | 2.1.0        |
| mini_magick               | 4.9.5        |
| rubyzip                   | 2.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.7.5        |
| bundler                   | 1.17.2       |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.6        |
| jwt                       | 2.1.0        |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| multi_json                | 1.13.1       |
| signet                    | 0.12.0       |
| httpclient                | 2.8.3        |
| google-api-client         | 0.38.0       |
| google-cloud-env          | 1.2.1        |
| google-cloud-core         | 1.3.1        |
| digest-crc                | 0.4.1        |
| google-cloud-storage      | 1.16.0       |
| jmespath                  | 1.4.0        |
| aws-partitions            | 1.296.0      |
| aws-eventstream           | 1.0.3        |
| aws-sigv4                 | 1.1.1        |
| aws-sdk-core              | 3.94.0       |
| aws-sdk-kms               | 1.30.0       |
| aws-sdk-s3                | 1.61.2       |
| CFPropertyList            | 3.0.1        |
| os                        | 1.0.1        |
| memoist                   | 0.16.0       |
| googleauth                | 0.9.0        |
| emoji_regex               | 2.0.0        |
| forwardable               | 1.2.0        |
| logger                    | 1.3.0        |
| stringio                  | 0.0.2        |
| ipaddr                    | 1.2.2        |
| openssl                   | 2.1.2        |
| ostruct                   | 0.1.0        |
| strscan                   | 1.0.0        |
| date                      | 2.0.0        |
| fileutils                 | 1.3.0        |
| io-console                | 0.4.7        |
| zlib                      | 1.0.0        |
| libxml-ruby               | 3.1.0        |
| rexml                     | 3.1.9        |
| psych                     | 3.1.0        |
| mutex_m                   | 0.1.0        |
| webrick                   | 1.4.2        |
</details>

*generated on:* **2020-10-09**
</details></pre>
</details>

**Comment by rogerluan** [comment]
Hi @danielgindi !

Please check my reply here and see if that helps 🙏  https://github.com/fastlane/fastlane/issues/17409#issuecomment-705842653

**Comment by danielgindi** [comment]
@rogerluan Dammit. I searched the issues but didn't find this one.
It's so weird that `gem` does not at least offer upgrades to other components in order to solve this. It's really that much of a "gem"...

Anyway, sadly this did not help. I've upgraded all dependencies the issue still exists.

The main issue is faraday. Fastlane depends apparently on an older version of faraday.

**Comment by rogerluan** [comment]
_fastlane_ actually depends on a newer version of it @danielgindi - by analyzinfg the `fastlane env` logs that you posted earlier, looks like you have `faraday-0.17.0, faraday-0.15.4`, and fastlane requires `faraday >= 1.0`. Try upgrading faraday on its own and retry updating _fastlane_ 🙏 For reference, these are my versions of the conflicting softwares:

```ruby
 faraday (1.0.1)
      google-api-client (>= 0.37.0, < 0.39.0)
      google-cloud-storage (>= 1.15.0, < 2.0.0)
    google-api-client (0.38.0)
      googleauth (~> 0.9)
    google-cloud-core (1.5.0)
      google-cloud-env (~> 1.0)
      google-cloud-errors (~> 1.0)
    google-cloud-env (1.3.3)
    google-cloud-errors (1.0.1)
    google-cloud-storage (1.29.1)
      google-api-client (~> 0.33)
      google-cloud-core (~> 1.2)
      googleauth (~> 0.9)
    googleauth (0.14.0)
```

**Comment by danielgindi** [comment]
> _fastlane_ actually depends on a newer version of it @danielgindi - by analyzinfg the `fastlane env` logs that you posted earlier, looks like you have `faraday-0.17.0, faraday-0.15.4`, and fastlane requires `faraday >= 1.0`. Try upgrading faraday on its own and retry updating _fastlane_ 🙏 For reference, these are my versions of the conflicting softwares:
> 
> ```ruby
>  faraday (1.0.1)
>       google-api-client (>= 0.37.0, < 0.39.0)
>       google-cloud-storage (>= 1.15.0, < 2.0.0)
>     google-api-client (0.38.0)
>       googleauth (~> 0.9)
>     google-cloud-core (1.5.0)
>       google-cloud-env (~> 1.0)
>       google-cloud-errors (~> 1.0)
>     google-cloud-env (1.3.3)
>     google-cloud-errors (1.0.1)
>     google-cloud-storage (1.29.1)
>       google-api-client (~> 0.33)
>       google-cloud-core (~> 1.2)
>       googleauth (~> 0.9)
>     googleauth (0.14.0)
> ```
> 
> PS: Once you install faraday 1.0.0+, uninstall older installations of faraday, to help reduce the chances of failure

I removed *all* packages mentioned, in all versions.  
The output stays exactly the same!

**Comment by andwaal** [comment]
Not solution to come with, but want to chime in that I am struggling with exact same issue as @danielgindi.  
Hopefully this can get fixed as an upgrade seem to be the fix for #17332

**Comment by Barnesst** [comment]
The folder where the studio is trying to install to may be the issue, You can always try to uninstall Android Studio and then install the version you.
